### PR TITLE
Support for ICC based on intel one api

### PR DESCRIPTION
**Pull request by omar-ocampo**
Installation default directory for intel one api is ~/intel/oneapi.
Added support to scons intelc tool to use intel one api path.

Test-tag: pr, -hw

Signed-off-by: Omar Ocampo <omar.ocampo.coronado@intel.com>

### CODE REVIEW

File: utils/sl/site_tools/intelc.py [hunk 1]
```diff
@@ -0,0 +1,657 @@
...
+            for d in glob.glob('/opt/intel/compilers_and_libraries_%s/linux' % version):
+                if os.path.exists(os.path.join(d, "bin", "ia32", "icc")) or os.path.exists(
+                        os.path.join(d, "bin", "intel64", "icc")):
+                    top = d
+                    break
+            return top
+
+        def find_in_intel_one_api_dir(version):
+            top = None
+            intel_one_api_path = os.path.expanduser("~/intel/oneapi")
```
Comment (jolivier23) [inline utils/sl/site_tools/intelc.py:414]: I think if you use user installed version, you should at least add /opt/intel/oneapi as the default place to look

**Review by jolivier23 — CHANGES_REQUESTED** [review]
I think we need to support /opt/intel installation first and foremost.   

Another area of exploration is whether we can do anything to wrap the installed intelc.py

File: utils/sl/site_tools/intelc.py (hunk 1 above)
Comment (omar-ocampo) [reply utils/sl/site_tools/intelc.py:414]: Ok, I'll change the default path. Is adding environment variables to modify where Scons look for icc, a "scons" thing to do?

**Comment by omar-ocampo** [comment]
> I think we need to support /opt/intel installation first and foremost.
> 
> Another area of exploration is whether we can do anything to wrap the installed intelc.py

Unsure how to wrap it, there is no class to extend from.

**Comment by omar-ocampo** [comment]
@jolivier23, should I fix the pylint issues? How can I re-run the tests which failed due to an unrelated change?

**Comment by jolivier23** [comment]
> @jolivier23, should I fix the pylint issues? How can I re-run the tests which failed due to an unrelated change?

it would be good to fix them.   Don't worry about tests for now but merging with latest master should help some.

**Review by jolivier23 — COMMENTED** [review]
Can you please move this to a draft if not ready for review.  For some reason, Github processed codeowners on a bunch of old reviews

**Pull request closed without merging**